src/magic_spell_effect.cpp

```diff
@@ -503,3 +509,245 @@ void spell_effect::translocate( const spell &sp, const Creature &caster,
+                   "thunder_near" );
+    int num_bolts = rng( 2, 4 );
+    for( int j = 0; j < num_bolts; j++ ) {
+        int xdir = 0;
+        int ydir = 0;
+        while( xdir == 0 && ydir == 0 ) {
+            xdir = rng( -1, 1 );
+            ydir = rng( -1, 1 );
+        }
+        int dist = rng( 4, 12 );
+        int boltx = target.x, bolty = target.y;
+        for( int n = 0; n < dist; n++ ) {
+            boltx += xdir;
+            bolty += ydir;
+            sp.create_field( {boltx, bolty, target.z} );
+            if( one_in( 4 ) ) {
+                if( xdir == 0 ) {
+                    xdir = rng( 0, 1 ) * 2 - 1;
+                } else {
+                    xdir = 0;
+                }
+            }
+            if( one_in( 4 ) ) {
+                if( ydir == 0 ) {
+                    ydir = rng( 0, 1 ) * 2 - 1;
+                } else {
+                    ydir = 0;
+                }
+            }
+        }
+    }
+}
+
+void spell_effect::fire_ball( const tripoint &target )
+{
+    explosion_handler::explosion( target, 180, 0.5, true );
+}
+
+void spell_effect::map( const spell &sp, const tripoint &target )
+{
+    player *p = g->critter_at<player>( target );
+    const bool new_map = overmap_buffer.reveal(
+                             point( target.x, target.y ), sp.aoe(), target.z );
+    if( new_map ) {
+        p->add_msg_if_player( m_warning, _( "You have a vision of the surrounding area..." ) );
+        p->mod_moves( -to_moves<int>( 1_seconds ) );
+    }
+}
+
+void spell_effect::blood( const spell &sp, const Creature &caster, const tripoint &target )
+{
+    bool blood = false;
+    for( const tripoint &tmp : g->m.points_in_radius( target, sp.aoe() ) ) {
+        blood |= sp.create_field( tmp ) && g->u.sees( tmp );
+    }
+    if( blood ) {
+        caster.add_msg_if_player( m_warning, _( "Blood soaks out of the ground and walls." ) );
+    }
+}
+
+void spell_effect::fatigue( const spell &sp, const Creature &caster, const tripoint &target )
+{
+    caster.add_msg_if_player( m_warning, _( "The fabric of space seems to decay." ) );
+    int x = rng( target.x - 3, target.x + 3 ), y = rng( target.y - 3, target.y + 3 );
+    sp.create_field( {x, y, target.z} );
+}
+
+void spell_effect::pulse( const spell &sp, const tripoint &target )
+{
+    sounds::sound( target, 30, sounds::sound_t::combat, _( "The earth shakes!" ), true, "misc",
+                   "earthquake" );
+    for( const tripoint &pt : g->m.points_in_radius( target, sp.aoe() ) ) {
+        g->m.bash( pt, sp.damage() );
+        g->m.bash( pt, sp.damage() );  // Multibash effect, so that doors &c will fall
+        g->m.bash( pt, sp.damage() );
+        if( g->m.is_bashable( pt ) && rng( 1, 10 ) >= 3 ) {
+            g->m.bash( pt, 999, false, true );
+        }
+    }
+}
+
+void spell_effect::entrance( const spell &sp, const tripoint &target )
+{
+    for( const tripoint &dest : g->m.points_in_radius( target, sp.aoe() ) ) {
```

Comment:
is there a particular reason you're using g->m.points_in_radius() instead of the spell effect area made specifically for spells?